e2eTests/e2eTestHelper.psm1

```diff
@@ -33,10 +33,21 @@ function SetTokenAndRepository {
         invoke-git config --global core.autocrlf false
     }
 
-    if ($appKey -and $appId) {
+    if (-not $github) {
+        # Running locally - Ensure the user is authenticated with the GitHub CLI.
+        # This is required for local runs to perform GitHub-related operations.
+        invoke-gh auth status
+    } elseif ($appKey -and $appId) {
+        # Running in GitHub Actions
         $token = @{ "GitHubAppClientId" = $appId; "PrivateKey" = ($appKey -join '') } | ConvertTo-Json -Compress -Depth 99
+    } else {
+        throw "GitHub App ID and Private Key not set. In order to run end to end tests, you need a Secret called E2E_PRIVATE_KEY and a variable called E2E_APP_ID."
     }
+
     # Repository isn't created yet so authenticating towards the .github repository
+    if (-not $github) {
+        gh auth refresh --scopes repo,admin:org,workflow,write:packages,read:packages,delete:packages,user,delete_repo

```

Comment:
Shouldn't that part be in `RefreshToken`?

Reply:
No, I think gh auth refresh is meant to be used when expanding the permission of the logged-in user. We only need to do that once. After that I assume gh cli keeps your account logged in with the same permissions.